### DISCUSSION TITLE
Type Mismatch Error in DSCEngineTest When Assigning Tuple from DeployDSC

### DISCUSSION THREAD

**Discussion by pherm0007** [Cyfrin/foundry-full-course-cu#2879] [Q&A]
I am encountering a type mismatch error in my Foundry test contract when attempting to destructure the return values of the run() function in my DSCEngineTest setup.

Type tuple(contract DecentralizedStableCoin, contract DSCEngine, contract HelperConfig) 
is not implicitly convertible to expected type tuple(contract DecentralizedStableCoin, contract DSCEngine, contract HelperConfig).
Here’s the relevant code snippet from the test contract:
<img width="1285" alt="Screenshot 2024-11-19 at 10 17 00" src="https://github.com/user-attachments/assets/584d24ba-577d-4665-b2af-098a3612cd2e">
What I Have Tried
Verified the run() function in DeployDSC and confirmed that it returns the tuple (DecentralizedStableCoin, DSCEngine, HelperConfig).
Ensured that all imported contracts are correct and align with their respective types.
Expected Behavior
The (dsc, engine, config) variables should be assigned the return values of deployer.run() without type mismatch issues.

**Comment by EngrPips** [comment]
Hello @pherm0007, Please show your `DeployDSC` contract.

**Reply by pherm0007** [reply]
here is it
<img width="1190" alt="deploydsctest" src="https://github.com/user-attachments/assets/9b04ecfe-8ee2-43c5-b1ef-acf56230e87e">

**Reply by EngrPips** [reply]
Hello @pherm0007, Please make your import command be the same in `DeployDSC` and in `DSCEngineTest` then let's see if the issue persists.

**Reply by pherm0007** [reply]
Thank you so much...It works

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.
